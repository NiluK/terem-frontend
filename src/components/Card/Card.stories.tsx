import React from 'react';
import Card from './Card';

export default {
  component: Card,
  title: 'Card',
};

export const Default = () => {
  return <Card img={'http://placehold.it/350x350'} title="Example" />;
};

export const withLocation = () => {
  return (
    <Card
      img={'http://placehold.it/350x350'}
      title="Example"
      location="Example Location"
    />
  );
};

export const withRestrictedImageWidth = () => {
  return (
    <Card
      img={'http://placehold.it/350x350'}
      imageWidth={50}
      title="Example"
      location="Example Location"
    />
  );
};

export const withRestrictedImageHeight = () => {
  return (
    <Card
      img={'http://placehold.it/350x350'}
      imageHeight={50}
      title="Example"
      location="Example Location"
    />
  );
};

export const withRestrictedImageWidthandHeight = () => {
  return (
    <Card
      img={'http://placehold.it/350x350'}
      imageWidth={50}
      imageHeight={40}
      title="Example"
      location="Example Location"
    />
  );
};
