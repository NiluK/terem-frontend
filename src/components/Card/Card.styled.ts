import styled from 'styled-components';

export const Wrapper = styled.div`
  display: inline-flex;
  background: ${(props): string => props?.theme?.colors?.card};
  border: 2px solid ${(props): string => props?.theme?.colors?.border};
`;

export const Image = styled.img`
  width: ${(props): string => (props.width ? `${props.width}px` : '100%')};
  object-fit: cover;
  height: ${(props): string => (props.height ? `${props.height}px` : '100%')};
`;

export const Info = styled.div`
  padding: 5px 10px 15px;
`;

export const Title = styled.h3`
  font-size: 12px;
  margin: 0;
  margin-bottom: 2px;
  color: ${(props): string => props?.theme?.colors?.primary};
`;

export const Location = styled.p`
  font-size: 12px;
  margin: 0;
  color: ${(props): string => props?.theme?.colors?.secondary};
  display: flex;
  align-items: flex-start;
`;
