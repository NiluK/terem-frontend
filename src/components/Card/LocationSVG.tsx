import React from 'react';
import styled from 'styled-components';

const SVG = styled.svg`
  fill: #7a91a2;
  height: 12px;
  width: 12px;
  margin-right: 2px;
`;

const LocationSVG = (): JSX.Element => {
  return (
    <SVG
      data-testid="location-svg"
      xmlns="http://www.w3.org/2000/svg"
      viewBox="0 0 24 24"
    >
      <path d="M12 0c-4.198 0-8 3.403-8 7.602 0 4.198 3.469 9.21 8 16.398 4.531-7.188 8-12.2 8-16.398 0-4.199-3.801-7.602-8-7.602zm0 11c-1.657 0-3-1.343-3-3s1.343-3 3-3 3 1.343 3 3-1.343 3-3 3z" />
    </SVG>
  );
};

export default LocationSVG;
