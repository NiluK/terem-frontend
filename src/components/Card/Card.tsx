import React from 'react';
import * as Styled from './Card.styled';
import LocationSVG from './LocationSVG';

interface Card {
  imageWidth?: number;
  imageHeight?: number;
  img: string;
  title: string;
  location?: string;
}

const Card = (props: Card): JSX.Element => {
  return (
    <Styled.Wrapper>
      <div>
        <Styled.Image
          width={props.imageWidth}
          height={props.imageHeight}
          src={props.img}
          alt={props.title}
        />
        <Styled.Info>
          <Styled.Title>{props.title}</Styled.Title>
          {props.location && (
            <Styled.Location>
              <LocationSVG />
              {props.location}
            </Styled.Location>
          )}
        </Styled.Info>
      </div>
    </Styled.Wrapper>
  );
};

export default Card;
