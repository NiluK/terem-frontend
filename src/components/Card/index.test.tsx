import React from 'react';

import { render } from '@testing-library/react';
import '@testing-library/jest-dom';

import Card from './Card';

describe('Card Structure', () => {
  const title = 'Hello World';
  const location = 'Hello Location';
  let container;
  beforeEach(() => {
    container = render(
      <Card
        title={title}
        img="https://placehold.it/320x320"
        location={location}
      />,
    );
  });

  it('matches snapshot', () => {
    expect(container).toMatchSnapshot();
  });

  it('renders title', () => {
    expect(container.getByText(title)).toBeInTheDocument();
  });
  it('renders image', () => {
    expect(container.getByAltText(title)).toBeInTheDocument();
  });
  it('renders location', () => {
    expect(container.getByText(location)).toBeInTheDocument();
  });
  it('renders svg location icon', () => {
    expect(container.queryByTestId('location-svg')).toBeInTheDocument();
  });
});

describe('Card Styles', () => {
  const title = 'Hello World';
  const location = 'Hello Location';

  it('renders image styles', () => {
    const container = render(
      <Card
        imageHeight={50}
        imageWidth={100}
        title={title}
        img="https://placehold.it/320x320"
        location={location}
      />,
    );
    expect(container.getByAltText(title)).toHaveStyle(`
      height: 50px;
      width: 100px
    `);
  });
});
