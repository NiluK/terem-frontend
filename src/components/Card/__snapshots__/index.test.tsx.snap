// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card Structure matches snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="Cardstyled__Wrapper-ke0us1-0 hsPdPI"
      >
        <div>
          <img
            alt="Hello World"
            class="Cardstyled__Image-ke0us1-1 jRiLjQ"
            src="https://placehold.it/320x320"
          />
          <div
            class="Cardstyled__Info-ke0us1-2 iIHUUS"
          >
            <h3
              class="Cardstyled__Title-ke0us1-3 efbcG"
            >
              Hello World
            </h3>
            <p
              class="Cardstyled__Location-ke0us1-4 jZRXLE"
            >
              <svg
                class="LocationSVG__SVG-sc-6iwt52-0 dNhYWG"
                data-testid="location-svg"
                viewBox="0 0 24 24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M12 0c-4.198 0-8 3.403-8 7.602 0 4.198 3.469 9.21 8 16.398 4.531-7.188 8-12.2 8-16.398 0-4.199-3.801-7.602-8-7.602zm0 11c-1.657 0-3-1.343-3-3s1.343-3 3-3 3 1.343 3 3-1.343 3-3 3z"
                />
              </svg>
              Hello Location
            </p>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="Cardstyled__Wrapper-ke0us1-0 hsPdPI"
    >
      <div>
        <img
          alt="Hello World"
          class="Cardstyled__Image-ke0us1-1 jRiLjQ"
          src="https://placehold.it/320x320"
        />
        <div
          class="Cardstyled__Info-ke0us1-2 iIHUUS"
        >
          <h3
            class="Cardstyled__Title-ke0us1-3 efbcG"
          >
            Hello World
          </h3>
          <p
            class="Cardstyled__Location-ke0us1-4 jZRXLE"
          >
            <svg
              class="LocationSVG__SVG-sc-6iwt52-0 dNhYWG"
              data-testid="location-svg"
              viewBox="0 0 24 24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M12 0c-4.198 0-8 3.403-8 7.602 0 4.198 3.469 9.21 8 16.398 4.531-7.188 8-12.2 8-16.398 0-4.199-3.801-7.602-8-7.602zm0 11c-1.657 0-3-1.343-3-3s1.343-3 3-3 3 1.343 3 3-1.343 3-3 3z"
              />
            </svg>
            Hello Location
          </p>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
